cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -717,7 +719,12 @@ public MapAndStatus(T loadMap, K status) {
             return onNew(reconciliation, host, apiClient, kafkaRebalance, rebalanceOptionsBuilder);
         } else {
             // Stay in the current NotReady state, returning null as next state
-            return Future.succeededFuture(new MapAndStatus<>(null, null));
+            Set<Condition> conditions = validate(reconciliation, kafkaRebalance);
+            if (!KafkaRebalanceState.NotReady.validateAnnotation(rebalanceAnnotation)) {
+                conditions.add(StatusUtils.buildWarningCondition("InvalidAnnotation", "Wrong annotation value: "
+                        + kafkaRebalance.getMetadata().getAnnotations() + ". The rebalance resource supports the following annotations in `NotReady` state: refresh"));
```

Comment:
If you want to futureproof this you can get the valid annotations from the KafkaRebalanceState enum itself by adding a `validStateString` method to the enum that returns a string of the valid annotations for that state? Then you can call that in the message. That way if you add new valid annotations to a state you don't have to change this string.

Reply:
Also this batch of code is basically the same in all the onState methods. Could you make it a generic method that just accepts a KafkaRebalanceState and the rebalanceAnnotation?

Reply:
Generic method inside the KRAO class itself, right?

Reply:
Yes, just a private method so you don't end up repeating the same code pattern over and over.